linkml/generators/fhirgen.py

```diff
@@ -0,0 +1,393 @@
+"""fhirgen: FHIR-related generators
+
+Currently, only supports Enum -> ValueSet
+
+Resources
+ 1. Enum def: https://linkml.io/linkml-model/docs/EnumDefinition/
+ 2. ValueSet def: https://www.hl7.org/fhir/valueset.html
+ 3. Issue:https://github.com/linkml/linkml/issues/905
+ 4. PR: https://github.com/linkml/linkml/pull/913
+
+TODO's
+ 1. R5: Need to see if it is different than R4 for ValueSet.
+ 2. Delete all stuff from file that I don't need
+ 3. Finish mapping__enum_valueset.csv
+   - Nested objects: e.g. FHIR's `identifier`
+   - Mutation dict of func/lambda's
+   - Do all of the 1..x cardinalities in FHIR first
+todo later's
+ 1. XML output
+"""
+import csv
+import json
+import os
+from copy import deepcopy
+# from enum import Enum
+from pathlib import Path
+from typing import Callable, Dict, Iterator, TextIO, Tuple, Union
+
+import click
+
+from linkml_runtime.dumpers import yaml_dumper
+from linkml_runtime.linkml_model.meta import ClassDefinition, ClassDefinitionName, Definition, DefinitionName, \
+    Element, EnumDefinition, SchemaDefinition, SlotDefinition, SubsetDefinition, TypeDefinition
+from linkml_runtime.utils.formatutils import camelcase, underscore
+from linkml_runtime.utils.schemaview import SchemaView
+# from linkml.utils.generator import Generator, shared_arguments
+# todo: Do I need _ensure_ranked? If so, move out of docgen?
+from linkml.generators.docgen import _ensure_ranked
+from linkml.utils.generator import Generator
+
+
+# todo: make some of these class properties?
+FHIRGEN_DIR = os.path.join(os.path.dirname(__file__), 'fhirgen')
+MODEL_MAPPING_PATH = os.path.join(FHIRGEN_DIR, 'mapping__enum_valueset.csv')
+FHIR_VALUESET_FIELD_MUTATIONS: Dict[str, Callable] = {
```

Comment:
FYI: This is how I handle mutations from one model field to the other, if needed. If it's complex enough, I'll use a function instead of a lambda.

Reply:
lambda should be fine.